spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/ReactorLoadBalancerClientAutoConfiguration.java

```diff
@@ -41,40 +50,91 @@
-			final List<WebClientCustomizer> customizers) {
-		return () -> {
-			for (WebClient.Builder webClientBuilder : getBuilders()) {
-				for (WebClientCustomizer customizer : customizers) {
-					customizer.customize(webClientBuilder);
+	@Configuration
+	@Conditional(OnNoRibbonDefaultCondition.class)
+	protected static class ReactorLoadBalancerExchangeFilterFunctionConfig {
+
+		private List<WebClient.Builder> webClientBuilders = Collections.emptyList();
+
+		List<WebClient.Builder> getBuilders() {
+			return this.webClientBuilders;
+		}
+
+		@Bean
+		public SmartInitializingSingleton loadBalancedWebClientInitializer(
+				final List<WebClientCustomizer> customizers) {
+			return () -> {
+				for (WebClient.Builder webClientBuilder : getBuilders()) {
+					for (WebClientCustomizer customizer : customizers) {
+						customizer.customize(webClientBuilder);
+					}
 				}
-			}
-		};
-	}
+			};
+		}
+
+		@Bean
+		public WebClientCustomizer loadBalancerClientWebClientCustomizer(
+				ReactorLoadBalancerExchangeFilterFunction filterFunction) {
+			return builder -> builder.filter(filterFunction);
+		}
+
+		@Bean
+		public ReactorLoadBalancerExchangeFilterFunction loadBalancerExchangeFilterFunction(
+				ReactiveLoadBalancer.Factory loadBalancerFactory) {
+			return new ReactorLoadBalancerExchangeFilterFunction(loadBalancerFactory);
+		}
+
+		@LoadBalanced
+		@Autowired(required = false)
+		void setWebClientBuilders(List<WebClient.Builder> webClientBuilders) {
+			this.webClientBuilders = webClientBuilders;
+		}
 
-	@Bean
-	public WebClientCustomizer loadBalancerClientWebClientCustomizer(
-			ReactorLoadBalancerExchangeFilterFunction filterFunction) {
-		return builder -> builder.filter(filterFunction);
 	}
 
-	@Bean
-	public ReactorLoadBalancerExchangeFilterFunction loadBalancerExchangeFilterFunction(
-			ReactiveLoadBalancer.Factory loadBalancerFactory) {
-		return new ReactorLoadBalancerExchangeFilterFunction(loadBalancerFactory);
+	private static final class OnNoRibbonDefaultCondition extends AnyNestedCondition {
+
+		private OnNoRibbonDefaultCondition() {
+			super(ConfigurationPhase.REGISTER_BEAN);
+		}
+
+		@ConditionalOnProperty(value = "spring.cloud.loadbalancer.ribbon.enabled",
+				havingValue = "false")
+		static class RibbonNotEnabled {
+
+		}
+
+		@ConditionalOnMissingClass("org.springframework.cloud.netflix.ribbon.RibbonLoadBalancerClient")
+		static class RibbonLoadBalancerNotPresent {
+
+		}
+
 	}
 
-	@LoadBalanced
-	@Autowired(required = false)
-	void setWebClientBuilders(List<WebClient.Builder> webClientBuilders) {
-		this.webClientBuilders = webClientBuilders;
+	private static class ReactorLoadBalancerClientRibbonWarnLogger {
+
+		private static final Log LOG = LogFactory
+				.getLog(ReactorLoadBalancerClientRibbonWarnLogger.class);
+
+		@PostConstruct
+		void logWarning() {
+			if (LOG.isWarnEnabled()) {
+				LOG.warn("You have RibbonLoadBalancerClient on your classpath. "
```

Comment:
Shouldnt we state the Ribbon is in maintenance mode and they should switch to the Spring Cloud implementation?